File: common/src/main/java/io/github/xiewuzhiying/vs_addition/compats/computercraft/peripherals/CheatCannonMountPeripheral.java
package io.github.xiewuzhiying.vs_addition.compats.computercraft.peripherals;

import dan200.computercraft.api.lua.LuaFunction;
import net.minecraft.core.BlockPos;
import net.minecraft.world.level.Level;
import rbasamoyai.createbigcannons.cannon_control.cannon_mount.CannonMountBlockEntity;

public class CheatCannonMountPeripheral extends CannonMountPeripheral{
    public CheatCannonMountPeripheral(String type, CannonMountBlockEntity tileEntity, Level level, BlockPos blockPos) {
        super(type, tileEntity, level, blockPos);
    }

    @LuaFunction(mainThread = true)
    public final void setPitch(double value){
        if(this.isRunning())
            this.tileEntity.setPitch((float) value);
    }

    @LuaFunction(mainThread = true)
    public final void setYaw(double value){
        if(this.isRunning())
            this.tileEntity.setYaw((float) value);
    }
}

File: common/src/main/kotlin/io/github/xiewuzhiying/vs_addition/compats/computercraft/peripheral/CheatCannonMountPeripheral.kt
package io.github.xiewuzhiying.vs_addition.compats.computercraft.peripheral

import dan200.computercraft.api.lua.LuaFunction
import rbasamoyai.createbigcannons.cannon_control.cannon_mount.CannonMountBlockEntity

open class CheatCannonMountPeripheral(peripheralType: String, tileEntity: CannonMountBlockEntity) : CannonMountPeripheral(peripheralType,
    tileEntity
) {
    @LuaFunction(mainThread = true)
    fun setPitch(value: Double) {
        if (this.isRunning()) tileEntity.setPitch(value.toFloat())
    }

    @LuaFunction(mainThread = true)
    fun setYaw(value: Double) {
        if (this.isRunning()) tileEntity.setYaw(value.toFloat())
    }
}